superset/viz.py

```diff
@@ -384,6 +384,25 @@ def query_obj(self):
         d['groupby'] = list(set(groupby) | set(columns))
         return d
 
+    def combine_metric(self, df):
```

Comment:
I think you can do all this in 2 lines of code using stack/unstack. https://pandas.pydata.org/pandas-docs/stable/reshaping.html#reshaping-by-stacking-and-unstacking